INPTPZ1/FractalSolver.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Numerics;
+using INPTPZ1.Mathematics;
+
+namespace INPTPZ1
+{
+	class FractalSolver
+	{
+		private const int MAX_ITERATIONS = 30;
+		private const double MIN_VALUE = 0.0001;
+		private const double ROOT_TOLERANCE = 0.01;
+		private const double PRECISION = 0.5;
+		private readonly string outputPath;
+		private PointF minimum;
+		private PointF maximum;
+		private readonly ImageHandler imageHandler;
+		private readonly ResolutionHolder resolutionHolder;
+		private Complex initialComplex;
+
+		public FractalSolver(ResolutionHolder resolutionHolder, PointF minimum, PointF maximum, string outputPath)
+		{
+			this.resolutionHolder = resolutionHolder;
+			imageHandler = new ImageHandler(resolutionHolder);
+			this.minimum = minimum;
+			this.maximum = maximum;
+			this.outputPath = outputPath;
+		}
+
+		internal void Solve()
+		{
+			List<Complex> rootList = new List<Complex>();
+
+			Polynom originalPolynom = InitOriginalPolynom();
+			Polynom derivedPolynom = originalPolynom.Derive();
+
+			double xstep = (maximum.X - minimum.X) / resolutionHolder.Width;
+			double ystep = (maximum.Y - minimum.Y) / resolutionHolder.Height;
+
+			for (int width = 0; width < resolutionHolder.Width; width++)
+			{
+				for (int height = 0; height < resolutionHolder.Height; height++)
+				{
+					double y = minimum.Y + width * ystep;
+					double x = minimum.X + height * xstep;
+
+					x = x == 0 ? MIN_VALUE : x;
+					y = y == 0 ? MIN_VALUE : y;
+
+					initialComplex = new Complex(x, y);
+
+					int iteration = FindSolutionByNewtonIteration(originalPolynom, derivedPolynom);
+
+					imageHandler.SetPixel(height, width, ImageHandler.GetColorByParams(iteration, GetRootCount(rootList)));
```

Comment:
this is not funny... You defined imageHandler's API and there are 'x' and 'y' parameters. Rewrite this whole method, because this is nonsense. You're mixing x with height and y with width... Test full algorithm on rectangular image!

Reply:
I didn't meant this as a joke. I simply haven't understood the previous remark considering this issue. I have retested the code with rectangular image and now it is clear to me. Thanks for your advice!